gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpRequestRunnable.java

```diff
@@ -109,30 +110,32 @@ public void run() {
       }
 
       // Populate HTTP path and query parameters.
-      Map<String, String> pathParams =
-          requestBuilder.getPathParams(request, methodDescriptor.getResourceNameField());
-      PathTemplate pathPattern = PathTemplate.create(methodDescriptor.endpointPathTemplate());
-      String relativePath = pathPattern.instantiate(pathParams);
-      GenericUrl url = new GenericUrl(endpoint + relativePath);
-      Map<String, List<String>> queryParams =
-          requestBuilder.getQueryParams(request, methodDescriptor.getQueryParams());
-      for (String queryParam : methodDescriptor.getQueryParams()) {
-        if (queryParams.containsKey(queryParam) && queryParams.get(queryParam) != null) {
-          url.set(queryParam, queryParams.get(queryParam));
+      GenericUrl url = new GenericUrl(endpoint + requestFormatter.getEndpointRelativePath(request));
+      Map<String, List<String>> queryParams = requestFormatter.getQueryParams(request);
+      for (String queryParam : queryParams.keySet()) {
+        if (queryParams.get(queryParam) != null) {
+          for (String val : queryParams.get(queryParam)) {
+            url.set(queryParam, val);
```

Comment:
If we get more than one item in the list, then this call will only use the last element in that list. 

https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/GenericUrl#set(java.lang.String,%20java.lang.Object)

Reply:
Oh didn't read up and see that GenericUrl is built on top of a regular Map<>. Using apache's URIBuilder now.